### PR TITLE
6326: fix bug reported at #6518

### PR DESCRIPTION
**Pull request by railway17**
### Details
Regress bugs that are displayed when testing with very long or wide images

### Fixed Issues
$ [Original Issue](https://github.com/Expensify/App/issues/6326)
$ [Regression Issue](https://github.com/Expensify/App/issues/6518)

### Tests
1. Prepare a very long image and send it to another.
2. Follow on the #6326 test case.
3. Prepare a very wide image and send it to another.
4. Follow on the #6326 test case.

### QA Steps
1. Prepare a very long image and send it to another.
2. Follow on the #6326 test case.
3. Prepare a very wide image and send it to another.
4. Follow on the #6326 test case.

### Tested On
- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web
[Web Regression Video](https://drive.google.com/file/d/174hcktknS6C3vGYvjwG96W09miaNVBKv/view?usp=sharing)

#### Desktop
[Desktop Regression Video](https://drive.google.com/file/d/1VKSbvWEXp7O3pvZVSOo8qtM7jeevtT2t/view?usp=sharing)

### CODE REVIEW

File: src/styles/styles.js
Comment (deetergp) [inline src/styles/styles.js:2315]: Would you please note what's going on in the UI in each of these cases?

**Review by deetergp — CHANGES_REQUESTED** [review]
Code looks good, but I had a couple comments regarding comments.

File: src/components/ImageView/index.js
Comment (railway17) [reply src/components/ImageView/index.js:89]: Actually, my internet speed is not good sometimes. And I noticed that it has some issue because we would not load image correctly.
If you don't need this, I will remove.

File: src/styles/styles.js
Comment (railway17) [reply src/styles/styles.js:2316]: I added the comments for each styling logic.
And noticed that my styling was not correct in the previous commit.
So, fixed it too. (when `isZoom === false && zoomScale < 1`)
Please review and let me know if something is still not clear.
Thank you

Comment (deetergp) [reply src/styles/styles.js:2316]: That's perfect. Thanks!

**Review by deetergp — DISMISSED** [review]

**Comment by deetergp** [comment]
Approved, but it looks like you've got a couple merge conflicts to resolve.

**Comment by railway17** [comment]
>

**Comment by railway17** [comment]
> Approved, but it looks like you've got a couple merge conflicts to resolve.

Oh.. sorry
I had mistake while checking this page.
Clicked the `close pull request` button unexpectedely.

**Comment by railway17** [comment]
> Approved, but it looks like you've got a couple merge conflicts to resolve.

Hi, @deetergp 
Do you want me to merge this PR to main by resolving conflicts myself?

**Comment by railway17** [comment]
Hi, @deetergp 
Because there was a problem resolving conflict in GitHub, I did it locally. (I noticed that my style code was moved to StyleUtil.js).
Please review it and let me know if you have any other problem
Thank you

File: src/components/ImageView/index.js
Comment (parasharrajat) [inline src/components/ImageView/index.js:89]: I didn't really understand what this comment is about?

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Could you please follow the template for the PR?
> $ Original Issue
> $ Regression Issue

Revert these to how they should be.

File: src/components/ImageView/index.js
Comment (railway17) [reply src/components/ImageView/index.js:89]: > 

For regression, there is new issue that is open.
Because I didn't know which issue number should be written, so I made it.
Which issue number should be written?
Original one?

Comment (railway17) [reply src/components/ImageView/index.js:89]: For this comment, I noticed that image loading is delayed because of internet speed.
In this case, image size is not decided even though the container (modal) is open.
Because of this reason, `zoomScale` is also not decided.
To avoid this problem, I wrote the code and comment like above.

Comment (parasharrajat) [reply src/components/ImageView/index.js:89]: > onLayout callback of the root View caused internet speed.

this is not making sense to me.

Please mention both issue URLs in the description.

Comment (railway17) [reply src/components/ImageView/index.js:89]: > 

It means 
1: You don't need such exception handling? Need to remove that code?
2. Should not write as `Original issue`, `Regression issue`?
Instead of it, should write urls directly like below?
`https://github.com/Expensify/App/issues/6326`, `https://github.com/Expensify/App/issues/6518`

Comment (railway17) [reply src/components/ImageView/index.js:89]: Hi, @parasharrajat 
Could you please let me know what you need exactly?

Comment (parasharrajat) [reply src/components/ImageView/index.js:89]: I will let others review it. I don't really know this part of code.

Comment (railway17) [reply src/components/ImageView/index.js:89]: If you want, I can remove that code shortly.
But I added that code because I noticed the issue in my side.
Please let me know whenever you have any feedback.

Comment (railway17) [reply src/components/ImageView/index.js:89]: Hi, @parasharrajat , @deetergp 
How are you doing?
Any feedback from other reviewers?
Please kindly let me know so that we can proceed.
Thank you

**Comment by parasharrajat** [comment]
I am not familiar with this part of code thus I would not be able to properly review it. I am also not assigned to this issue. Please continue without my review. 

I would recommend to get more eyes on this PR as there could be more regressions.

**Comment by railway17** [comment]
> I am not familiar with this part of code thus I would not be able to properly review it. I am also not assigned to this issue. Please continue without my review.
> 
> I would recommend to get more eyes on this PR as there could be more regressions.

@parasharrajat 
Thank you for your quick reply.


> Approved, but it looks like you've got a couple merge conflicts to resolve.

@deetergp 
Could you approve this PR again?
You have already approved this PR but it needs another approval while I merge with `main` in my local.
I think it is caused by my [bad activity](https://github.com/expensify/App/pull/6535#issuecomment-984192014).
Please check my merging (I merged correctly but you can double-check) and approve so that we can proceed.
Thank you

**Comment by railway17** [comment]
@roryabraham 
What means your assignment?
Did something go wrong?

**Comment by roryabraham** [comment]
Hi @railway17. I'm going to review this – this is time-sensitive so let's work together to get this merged ASAP.

**Comment by railway17** [comment]
> Hi @railway17. I'm going to review this – this is time-sensitive so let's work together to get this merged ASAP.

Ok, here is 1:00am now, but I will try to work with you together.
Let's start now.

File: src/components/ImageView/index.js
Comment (roryabraham) [inline src/components/ImageView/index.js:93]: Why not just use the `newZoomScale` in all cases?

File: src/components/ImageView/index.js
Comment (roryabraham) [reply src/components/ImageView/index.js:89]: I tried hard to understand what you were trying to communicate with this comment, but it doesn't make sense to me. It sounds like there's some race condition at play here? If you can explain what you mean in a different way, that would be great. Otherwise, please just remove this comment.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Also, what if the window resizes? I think maybe we need to handle this by:

1. Saving the image height as a class field before calling `setImageRegion`
1. Calling `setImageRegion` again in `componentDidUpdate` if the window dimensions changed.

**Comment by roryabraham** [comment]
@railway17 It's okay – what I'm going to do instead is revert your [original PR](https://github.com/Expensify/App/pull/6442) and that way we can take the time we need to re-implement it without causing a regression.

**Comment by railway17** [comment]
> @railway17 It's okay – what I'm going to do instead is revert your [original PR](https://github.com/Expensify/App/pull/6442) and that way we can take the time we need to re-implement it without causing a regression.

I removed the comment from that code and pushed it.
It is under checking now.

**Comment by roryabraham** [comment]
> I removed the comment from that code and pushed it.
It is under checking now.

Okay, I had a couple more comments in my last review

File: src/components/ImageView/index.js
Comment (railway17) [reply src/components/ImageView/index.js:92]: Actually, I was going to set zoomScale in the `onLayout` event of the parent view.
But while testing in my local, I noticed that it is sometimes 0 because image loading is slower than view loading.
So, reset with newZoomScale here.

**Comment by railway17** [comment]
> 1. setImageRegion



> 2\. setImageRegion
Actually, [this issue](https://github.com/Expensify/App/issues/6326)  requirement was fixing cursor icon issue while zoomin or zoomout for small thumbnails.
So, I didn't consider about window resize.

**Review by roryabraham — APPROVED** [review]
Okay, I'm going to approve and CP this to unblock the deploy, but will follow up with a small PR of my own to address my comments.

**Comment by botify** [comment]
@roryabraham looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by roryabraham** [comment]
Not an emergency, just CP'ing to speed up deploys. Only change since last successful E2E test was the removal of a comment.

**Comment by parasharrajat** [comment]
We have found another [regression](https://github.com/Expensify/App/pull/6535) from this PR. @railway17 

cc: @roryabraham

**Pull request merged**
